test/test_diffusion.py

```diff
@@ -43,14 +43,57 @@
     diffusion_flux,
     diffusion_operator,
     grad_facial_flux_weighted,
+    PrescribedFluxDiffusionBoundary,
     DirichletDiffusionBoundary,
-    NeumannDiffusionBoundary)
+    NeumannDiffusionBoundary,
+    RobinDiffusionBoundary)
+from mirgecom.integrators import rk4_step
 from mirgecom.simutil import get_box_mesh
 from mirgecom.discretization import create_discretization_collection
 
 logger = logging.getLogger(__name__)
 
 
+def test_diffusion_boundary_conditions(actx_factory):
+    """Checks the boundary conditions for diffusion operator."""
+    actx = actx_factory()
+    dim = 2
+    n = 4
+    order = 4
+
+    mesh = get_box_mesh(dim, -1, 1, n)
+    dcoll = create_discretization_collection(actx, mesh, order)
+    nodes = actx.thaw(dcoll.nodes())
+
+    u_ref = -1.5
+    alpha = 0.5
+    kappa = 2
+    u = 0.5*nodes[0] + 1.0
+
+    def make_dirichlet_bc(btag):
+        bdry_u = op.project(dcoll, "vol", BoundaryDomainTag(btag), u)
+        return DirichletDiffusionBoundary(bdry_u)
+
+    boundaries = {
+        BoundaryDomainTag("-1"): RobinDiffusionBoundary(u_ref, alpha),
+        BoundaryDomainTag("+1"): NeumannDiffusionBoundary(0.5),
+        BoundaryDomainTag("-2"): make_dirichlet_bc("-2"),
+        BoundaryDomainTag("+2"): PrescribedFluxDiffusionBoundary(0.0),
+    }
+
+    rhs, grad_u = diffusion_operator(dcoll, kappa=kappa, u=u,
+            boundaries=boundaries, return_grad_u=True,
+            gradient_numerical_flux_func=grad_facial_flux_weighted)
+
+    err_grad_x = actx.to_numpy(op.norm(dcoll, grad_u[0] - 0.5, np.inf))
+    err_grad_y = actx.to_numpy(op.norm(dcoll, grad_u[1] - 0.0, np.inf))
+    assert err_grad_x < 1.e-9
+    assert err_grad_y < 1.e-9
+
+    err_rhs = actx.to_numpy(op.norm(dcoll, rhs, np.inf))
+    assert err_rhs < 1.e-8
+
+
```

Comment:
Any chance you could add a case to `test_diffusion_accuracy` with a Robin BC? It would be good to check that it converges with the right order.

Reply:
I added a test, but it is interesting/confusing that the error in test converges with "order" while the example I ran (see PR description) converges with order (for odd) and order + 1 (for even).

Thoughts/ideas?

Reply:
My guess is it just comes down to computing the error on an expression that includes derivatives of the solution rather than just the solution itself. I set up another test here: https://github.com/majosm/mirgecom/commit/4a99f4e8562c6df78e21010f666cffe977b6fb4b. It checks the solution on the volume, and it gets the expected convergence rate. When I modify it to compute the error of `grad(u)`, it tends to be 1 order lower. I did notice that if I don't run for long enough, it can appear to converge at a higher order; maybe this is what you observed in your example case?